tests/python/unittest/test_kvstore.py

```diff
@@ -76,7 +76,7 @@ def check_row_sparse_pull(kv, count):
         for i in range(count):
             vals.append(mx.nd.zeros(shape).tostype('row_sparse'))
             row_id = np.random.randint(num_rows, size=num_rows)
-            row_ids.append(mx.nd.array(row_id))
+            row_ids.append(mx.nd.array(row_id).reshape((2,2)))
```

Comment:
Why reshape?
One suggestion: (2, 2) is too hard-coded. If shape is changed in the beginning, the test would fail.

Reply:
You are right.. Made it less hard-coded by calculating the dimension based on total number of elements. 
I'm testing 2-D rowids explicitly because that seems to be a common use case, many users forget to reshape it to 1-D before passing it to kvstore. It is now supported with this PR.